opt_einsum/helpers.py

```diff
@@ -117,11 +117,7 @@ def find_contraction(positions, input_sets, output_set):
     """
 
     remaining = list(input_sets)
-    idx_contract = set()
-
-    for i in sorted(positions, reverse=True):
-        idx_contract |= remaining.pop(i)
-
+    idx_contract = set.union(*map(remaining.pop, sorted(positions, reverse=True)))
```

Comment:
Do we get much of a performance increase by rewriting this term as shown? Most of them are ok, but this one takes a bit to understand.